homeassistant/components/onkyo/config_flow.py

```diff
@@ -165,6 +167,50 @@ async def async_step_eiscp_discovery(
             ),
         )
 
+    async def async_step_ssdp(
+        self, discovery_info: ssdp.SsdpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle flow initialized by SSDP discovery."""
+        _LOGGER.debug("Config flow start ssdp: %s", discovery_info)
+
+        udn = discovery_info.ssdp_udn
+        if udn:
+            udn_parts = udn.split(":")
+            if len(udn_parts) == 2:
+                uuid = udn_parts[1]
+                last_uuid_section = uuid.split("-")[-1].upper()
+                await self.async_set_unique_id(last_uuid_section)
+                self._abort_if_unique_id_configured()
+
+        if discovery_info.ssdp_location is None:
+            _LOGGER.error("SSDP location is None")
+            return self.async_abort(reason="unknown")
+
+        host = urlparse(discovery_info.ssdp_location).hostname
+
+        if host is None:
+            _LOGGER.error("SSDP host is None")
+            return self.async_abort(reason="unknown")
+
+        try:
+            info = await async_interview(host)
```

Comment:
Its unfortunate this API raises exceptions and can also return `None`.  Ideally it would always raise when `None` is returned with a reason so `if info is None:` would not need to be checked.   OK to merge as-is though

Reply:
I left it as is for now